src/portfft/common/transfers.hpp

```diff
@@ -488,6 +488,250 @@ PORTFFT_INLINE void local2global(detail::global_data_struct<1> global_data, Loca
       global_data, global, local, total_num_elems, global_offset, local_offset);
 }
 
+/**
+ * Driver function to copy data between local and global memory when data is in PACKED format in both, local
+ * as well as global memory, when the storage scheme is INTERLEAVED_COMPLEX
+ *
+ * @tparam Group Group level taking part in the copy, should be one of level::SUBGROUP or level::WORKGROUP
+ * @tparam Direction Direction of the copy, expected to be either transfer_direction::LOCAL_TO_GLOBAL or
+ * transfer_direction::GLOBAL_TO_LOCAL
+ * @tparam SubgroupSize Subgroup Size
+ * @tparam LocView Type of view of the local memory
+ * @tparam T Scalar Type
+ * @param global_ptr Pointer to the input / output global memory
+ * @param loc_view Local memory view containing the input
+ * @param global_offset Offset to be applied to the input / output pointer
+ * @param local_offset Offset to be applied to local memory view
+ * @param n_elements_to_copy Number of scalar elements to copy
+ * @param global_data global_data_struct associated with the kernel launch
+ */
+template <detail::level Group, detail::transfer_direction Direction, Idx SubgroupSize, typename LocView, typename T>
+PORTFFT_INLINE void local_global_packed_copy(T* global_ptr, LocView& loc_view, IdxGlobal global_offset,
```

Comment:
Most of these new transfer functions are refactoring that - as far as I can tell - is not required for the Bluestein implementation. I'd prefer it if they could be reverted, so that we could discuss the course we want to take with them in a separate PR in the future.

Reply:
I could not have done Bluestein without the refactoring, as it otherwise would have been very difficult to debug. The number of lines in subgroup_impl was reduced significantly, whereas previously I found myself scrolling for some time and as it turned out I had only traversed a small fraction of the entire kernel ! 

Also, quite a lot of refactoring was necessary . For example in the sg_bluestein functions, when I need to load/store from/to local memory, and how sg_bluestein_packed_copy uses local_global_packed_copy.
Without the refactor, this would have resulted in a massive amount of duplicated code (repeated view creations)

Reply:
Also, we can keep perfecting the refactor by taking it further from here later on whenever we come back to it, but at least it leaves the code in a  usable state. I have applied the same to workgroup level bluestein as well

Reply:
I still think this should be a separate PR, even if the code might look ugly.
* On principal, 1 PR should only do 1 thing. 
* We used to have lots of transfer functions, do we want that again?
* Separating out functions like this makes it harder to unify the implementation later.

Reply:
> On principal, 1 PR should only do 1 thing.

...Ideally, assuming given sufficient time and all the resources (including the ease of implementation) available to implement the goal of the PR. This is also why I have always advocated for a refactor as you go approach before putting up the PR

> We used to have lots of transfer functions, do we want that again?

 If anything they make the code more verbose, and you can read the order our operations as if reading phrases of English, i,e, copy_global_to_local ---> local_to_private --> compute --> private_to_local ---> local_to_global. All the complexity involved with how many views to be created is shifted elsewhere. We can keep trickling this down to our other sections later on, and adjust if required. 

Sorry for taking a firm stance here, but I do not suppose I will be reverting this change, given the time crunch and amount of things I would need to change, and no downside associated with it

Reply:
Just catching up on this comment, I think Hugh made good points here. We're also trying to push for splitting the PRs as much as possible and discussing early when a large change is needed. Given that we have already refactored transfer functions multiple times we would have needed long discussions which we don't have time for anymore.

I strongly suggest that we don't bother merging Bluestein in main but instead:

- Keep all the Bluestein work in a bluestein branch.
- Create and close the other 2 PRs for Bluestein that you had plan if they are ready already. The could be useful if we try to merge this in main again in the future.